src/datadog/trace_sampler_config.h

```diff
@@ -28,7 +28,7 @@ struct TraceSamplerConfig {
 
   Optional<double> sample_rate;
```

Comment:
Default value should be `1.0` right? How does it work today?

Reply:
Almost:

No rate => no synthesized rule.

No rules => default sampling mechanism => Agent-provided rates, or `Rate::one()` otherwise.